core-common/src/main/java/org/zalando/nakadi/util/JsonPathAccess.java

```diff
@@ -27,12 +27,12 @@ public Object get(final String path) throws InvalidPartitionKeyFieldsException {
 
         while ((field = pathTokenizer.nextToken()) != null) {
             if (!(curr instanceof JSONObject)) {
```

Comment:
it is not clear why you do check here, although you have JSONObject for sure from constructor and when getting the field.

Reply:
While getting the field, we cast the `curr` object:
```
curr = ((JSONObject) curr).get(field);
```
However, the received `field`, (new current) may bot be instance of `JSONObject`.

Reply:
you are right, I missed that. just discard my comment.